Allow systemd-sleep transition to appropriate domains

zpytela opened the pull request:
The /usr/lib/systemd/system-sleep directory is a place for local
scripts/binaries, executed immediately before entering and after
leaving system suspend/hibernation.

Resolves: rhbz#1927551

WOnder93 commented on policy/modules/system/systemd.fc:
```diff
@@ -50,6 +50,7 @@ HOME_DIR/\.config/systemd/user(/.*)?		gen_context(system_u:object_r:systemd_unit
 /usr/lib/systemd/system/.*sleep.*\.(service|target)		--	gen_context(system_u:object_r:power_unit_file_t,s0)
 /usr/lib/systemd/system/.*shutdown.*\.(service|target)		--	gen_context(system_u:object_r:power_unit_file_t,s0)
 /usr/lib/systemd/system/.*suspend.*\.(service|target)		--	gen_context(system_u:object_r:power_unit_file_t,s0)
+/usr/lib/systemd/system-sleep(/.*)		gen_context(system_u:object_r:systemd_sleep_exec_t,s0)
```
The `()`s are not needed. And there should be a `--` so that directories are not matched.

Also, I'd move it one line below so that it's not in the middle of unit file fcontexts.

WOnder93 commented on policy/modules/system/systemd.fc:
```diff
@@ -51,6 +51,7 @@ HOME_DIR/\.config/systemd/user(/.*)?		gen_context(system_u:object_r:systemd_unit
 /usr/lib/systemd/system/.*shutdown.*\.(service|target)		--	gen_context(system_u:object_r:power_unit_file_t,s0)
 /usr/lib/systemd/system/.*suspend.*\.(service|target)		--	gen_context(system_u:object_r:power_unit_file_t,s0)
 /usr/lib/systemd/system/systemd-userdbd\.(service|socket)		--	gen_context(system_u:object_r:systemd_userdbd_unit_file_t,s0)
+/usr/lib/systemd/system-sleep(/.*)?		gen_context(system_u:object_r:systemd_sleep_exec_t,s0)
```
I'm not convinced that we want to label also the directory with the `systemd_sleep_exec_t` type... At the very least, `systemd_sleep_t` will now need new permissions to read the directory.

Also, won't this cause a regression for the tlp script? Depending on what it does exactly, systemd_sleep_t might not be a good label for it.

Ideally, each service that installs some script there would have its own "sleep script" domain and file label and would register a transition from `systemd_sleep_t` to this domain. And for custom user scripts there could be a special customizable type (see `customizable_types(5)`) with a transition to some near-unconfined domain (or just `unconfined_service_t`) so they can get their scripts to work easily. Otherwise we'll be playing an infinite catch-up game with what users put there and will want to work and will end up with bloated policy for `systemd_sleep_t`...

zpytela replied:
I changed the PR especially for `/usr/lib/systemd/system-sleep` is expected to be a place for user scripts.

zpytela committed:
Allow systemd-sleep transition to unconfined_service_t on bin_t executables

The /usr/lib/systemd/system-sleep directory is a place for local
scripts/binaries, executed immediately before entering and after
leaving system suspend/hibernation. Files in the directory will have
the generic bin_t type unless specified differently in the policy.
Note systemd considers these scripts hack and rather recommends using
the Inhibitor interface.
https://www.freedesktop.org/wiki/Software/systemd/inhibit

Resolves: rhbz#1927551

zpytela committed:
Allow systemd-sleep transition to tlp_t

Scripts in the /usr/lib/systemd/system-sleep directory are executed
for the system suspend, hibernation, or hybrid hibernation.
Scripts with the generic bin_t type will transition to unconfined_service_t,
but confined scripts like tlp can transition to the tlp_t confined
domain.

zpytela committed:
Allow systemd-sleep transition to sysstat_t

Scripts in the /usr/lib/systemd/system-sleep directory are executed
for the system suspend, hibernation, or hybrid hibernation.
While scripts with the generic bin_t type will transition to
unconfined_service_t, sysstat.sleep script executes the confined
/usr/lib64/sa/sa1 executable, so it can transition to the sysstat_t
domain.
